internal/exporter/exporter.go

```diff
@@ -60,7 +60,7 @@ func (v *exporter) RecordMetrics(interval time.Duration) {
 func (v *exporter) recordingIteration(logPath string) {
 	directories, err := listDirs(logPath)
 	if err != nil && v.Logger != nil {
-		v.Logger.Error("Error listing directories", slog.Any("err", err))
+		v.Logger.Error("Error listing directories", slog.Any("err", err)) //nolint:noctx // no need for context here
```

Comment:
Let's introduce ctx arg in RecordMetrics and pass it to recordingIteration

Reply:
please check again